account_invoice_mass_sending/models/account_move.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.move"
+
+    sending_in_progress = fields.Boolean(
+        default=False,
+        help="If checked, the invoice is already being processed, "
+        "and it will prevent the sending of a duplicated mail.",
+    )
+
+    def mass_send_print(self):
+        """
+        This method triggers the asynchronous "Send & Print" for the selected
+        invoices for which there is no asynchronous sending in progress.
+        """
+        invoices_to_send = self.filtered(lambda i: not i.sending_in_progress)
+        in_progress_invoices_count = len(self) - len(invoices_to_send)
+        title = _("Invoices: Send & Print")
+        if in_progress_invoices_count > 0:
+            warn_msg = _(
+                "The sending of {in_progress_invoices_count} invoices is "
+                "already in progress."
+            ).format(in_progress_invoices_count=in_progress_invoices_count)
+            self.env.user.notify_info(title=title, message=warn_msg)
+        if invoices_to_send:
+            msg = _(
+                "The sending of {invoices_count} invoices will be processed "
+                "in background."
+            ).format(invoices_count=len(invoices_to_send))
+            self.env.user.notify_info(title=title, message=msg)
+            invoices_to_send.write({"sending_in_progress": True})
+            invoices_to_send.with_delay().do_prepare_send_print()
+
+    def do_prepare_send_print(self):
+        for rec in self:
+            rec.with_delay().do_send_print()
+
+    def do_send_print(self):
+        for rec in self:
+            if not rec.partner_id.email:
+                raise UserError(
+                    _("Missing email address on customer " "'{customer_name}'.").format(
```

Comment:
The syntax can be simplified, maybe there was two lines before.

Reply:
done :)